src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file bayesian_ridge.cpp
+ * @author Clement Mercier 
+ *
+ * Implementation of Bayesian Ridge regression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "bayesian_ridge.hpp"
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/core/util/timers.hpp>
+
+using namespace mlpack;
+using namespace mlpack::regression;
+
+
+BayesianRidge::BayesianRidge(const bool centerData,
+                             const bool scaleData,
+                             const int nIterMax,
+                             const double tol) :
+  centerData(centerData),
+  scaleData(scaleData),
+  nIterMax(nIterMax),
+  tol(tol)
+{/* Nothing to do */}
+
+double BayesianRidge::Train(const arma::mat& data,
+                            const arma::rowvec& responses)
+{
+  Timer::Start("bayesian_ridge_regression");
+
+  arma::mat phi;
+  arma::rowvec t;
+  arma::colvec eigval;
+  arma::mat eigvec;
+  arma::colvec eigvali;
+
+  // Preprocess the data. Center and scale.
+  responsesOffset = CenterScaleData(data,
+                                    responses,
+                                    centerData,
+                                    scaleData,
+                                    phi,
+                                    t,
+                                    dataOffset,
+                                    dataScale);
+
+  // Compute this quantities once and for all.
+  const arma::colvec vecphitT = phi * t.t();
+
+  // Enforce symmetry of the covariance matrix before eig_sym.
+  const arma::mat phiphiT =  arma::symmatu(phi * phi.t());
+
+  if (arma::eig_sym(eigval, eigvec, phiphiT) == false)
+  {
+    Log::Warn << "Eigen Decomposition failed as Eigen Value "
+              << "does not exists ." << std::endl;
```

Comment:
```suggestion
    Log::Warn << "BayesianRidge::Train(): Eigendecomposition of covariance failed!" << std::endl;
```

I don't understand the reasoning `eigenvalue does not exists`, maybe it's better to leave the reasoning out and just say that it failed?

Reply:
Hi @rcurtin, you are totally right, my mistake.